dart/lib/src/sentry_tracer.dart

```diff
@@ -197,6 +205,9 @@ class SentryTracer extends ISentrySpan {
       return NoOpSentrySpan();
     }
 
+    // reset the timer if a new child is added
+    _scheduleTimer();
```

Comment:
m: Cocoa uses I different approach that I believe makes more sense.
Instead o reseting the timer, we cancel it all, and every time all child spans are finished we start the timer again to see if something else will happen.

Some operations may take more than the default timeout, together with `waitForChildren` what we really need to do is give a little room to something happen after an operation is finished.

Reply:
This is not a blocker. 
Im just suggestion something I believe make more sense. 
We can keep the current implementation.